### PR TITLE
ANDROID-15230 Fix suppressed long method

### PR DESCRIPTION
**Pull request by jeprubio**
### :goal_net: What's the goal?
Fix suppress LongMethod

### :construction: How do we do it?
- Remove the suppressions of LongMethod.
- Added some private methods to make the previous long methods smaller.
- Scoutting: Removing the annotation ExperimentalPagerApi which was then suppressed in the same file with the OptIn.

### :test_tube: How can I test this?
- [x] Video checking the updated code of lists and progress buttons:

https://github.com/user-attachments/assets/9db92577-046d-4e3a-b09e-32fc22517535

- [x] [Catalog download link](https://install.appcenter.ms/orgs/tuenti-organization/apps/mistica-catalog/distribution_groups/public/releases/640)

### CODE REVIEW

File: library/src/main/java/com/telefonica/mistica/button/ProgressButton.kt [hunk 1]
```diff
@@ -127,88 +201,37 @@ class ProgressButton : FrameLayout {
...
+    private fun addViews() {
         addView(buttonBackground)
         addView(buttonNormal)
         addView(buttonLoading)
         addView(progressBar)
-
-        setVisibilityAndColors()
-    }
-
-    fun getText(): CharSequence =
```
Comment (jeprubio) [inline library/src/main/java/com/telefonica/mistica/button/ProgressButton.kt:160]: moved the public methods on top of the private ones I've created

File: library/src/main/java/com/telefonica/mistica/button/ProgressButton.kt [hunk 2]
```diff
@@ -66,8 +66,72 @@ class ProgressButton : FrameLayout {
...
-    @Suppress("LongMethod")
     private fun init(attrs: AttributeSet? = null, defStyleAttr: Int = 0) {
+        setupAttributes(attrs, defStyleAttr)
+        setupViewProperties()
+        setupButtonNormal()
+        setupButtonLoading()
+        setupProgressBar()
+        setButtonBackground()
+        addViews()
+        setVisibilityAndColors()
```
Comment (jeprubio) [inline library/src/main/java/com/telefonica/mistica/button/ProgressButton.kt:77]: Splitted the code in private methods

File: library/src/main/java/com/telefonica/mistica/compose/carousel/CarouselPagerIndicator.kt [hunk 3]
```diff
@@ -156,8 +154,6 @@ fun CarouselPagerIndicator(
         indicatorShape = indicatorShape)
 }
 
-@ExperimentalPagerApi
```
Comment (jeprubio) [inline library/src/main/java/com/telefonica/mistica/compose/carousel/CarouselPagerIndicator.kt:159]: These Experimental annotations are wrong, they were hidden by the OptIn on the top but they shouldn't have been there.

File: library/src/main/java/com/telefonica/mistica/button/ProgressButton.kt [hunk 4]
```diff
@@ -127,88 +201,37 @@ class ProgressButton : FrameLayout {
...
                         val outsideBounds =
                             event.x < 0 || event.y < 0 || event.x > view.measuredWidth || event.y > view.measuredHeight
                         buttonNormal.isPressed = !outsideBounds
                         buttonLoading.isPressed = !outsideBounds
                     }
+
                     MotionEvent.ACTION_UP, MotionEvent.ACTION_CANCEL -> {
                         buttonNormal.isPressed = false
                         buttonLoading.isPressed = false
+                        view.performClick()
```
[the remark this answers is not in the document]
Comment (jeprubio) [reply library/src/main/java/com/telefonica/mistica/button/ProgressButton.kt:222]: Detekt marks it as error if view.performClick() is not invoked on a setOnTouchListener lambda

[the remark this answers is not in the document]
Comment (yamal-alm) [reply library/src/main/java/com/telefonica/mistica/button/ProgressButton.kt:222]: I'm worried about any unexpected behavior and if it wasn't added before for a reason.. but I guess the button is still working as before right?

[the remark this answers is not in the document]
Comment (jeprubio) [reply library/src/main/java/com/telefonica/mistica/button/ProgressButton.kt:222]: Yup, I made a video in the pr description which checks the 2 updated elements (the button and the list)

**Review by yamal-alm — APPROVED** [review]
thanks!

**Review by DevPabloGarcia — APPROVED** [review]
Thanks!

**Pull request merged**